ambry-store/src/main/java/com/github/ambry/store/IndexSegment.java

```diff
@@ -109,6 +112,7 @@
   private Offset prevSafeEndPoint = null;
   // reset key refers to the first StoreKey that is added to the index segment
   private Pair<StoreKey, PersistentIndex.IndexEntryType> resetKey = null;
+  private short resetKeyLifeVersion = UNINITIALIZED_RESET_KEY_VERSION;
```

Comment:
I think it's better to create a class to represent the reset key info, Something like
```
public class ResetKeyInfo {
    private final StoreKey storeKey;
    private final IndexEntryType keyType;
    private final short lifeVersion;
}
```
The resetKey isn't used anywhere, we can assume that all three pieces of info will be required by the caller all together.

Reply:
yeah, it's a good way to wrap them up and would be more extensible if we add new fields in the future.